pkg/autoscaler/metrics/stats_scraper_test.go

```diff
@@ -535,6 +537,179 @@ func TestScrapeDoNotScrapeIfNoPodsFound(t *testing.T) {
 	}
 }
 
+func TestMixedPodShuffle(t *testing.T) {
+	ctx, cancel, informers := SetupFakeContextWithCancel(t)
+	wf, err := controller.RunInformers(ctx.Done(), informers...)
+	if err != nil {
+		cancel()
+		t.Fatal("StartInformers() =", err)
+	}
+	t.Cleanup(func() {
+		cancel()
+		wf()
+	})
+
+	tm := metav1.NewTime(time.Now().Add(-time.Hour))
+	const numPods = 25
+	makePods(ctx, "old-", 5, tm)
+	makePods(ctx, "new-", numPods-5, metav1.Now())
+	wantScrapes := int(populationMeanSampleSize(numPods))
+	t.Log("WantScrapes", wantScrapes)
+
+	client := newTestScrapeClient(testStats, []error{nil})
+	scraper, err := serviceScraperForTest(ctx, t, client, client, true)
+	if err != nil {
+		t.Fatalf("serviceScraperForTest=%v, want no error", err)
+	}
+
+	_, err = scraper.Scrape(defaultMetric.Spec.StableWindow)
+	if err != nil {
+		t.Fatal("scraper.Scrape() returned error:", err)
+	}
+	if got, want := len(client.urls), wantScrapes; got != want {
+		t.Errorf("Got = %d unique URLS, want: %d", got, want)
```

Comment:
Should we fatal if the number doesn't match? Because the verification below will fail too.

Same elsewhere.

Reply:
Sure.